pkg/validation/validation.go

```diff
@@ -207,3 +212,15 @@ func gatewayAddresses(gw *gatewayv1alpha1.Gateway) error {
 	}
 	return nil
 }
+
+// gatewayContour returns the contour associated with gw, if valid.
+func gatewayContour(ctx context.Context, cli client.Client, gw *gatewayv1alpha1.Gateway) (*operatorv1alpha1.Contour, error) {
+	contour, err := objgw.ContourForGateway(ctx, cli, gw)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get contour for gateway %s/%s", gw.Namespace, gw.Name)
+	}
+	if contour.Spec.Namespace.Name != gw.Namespace {
+		return nil, fmt.Errorf("invalid contour.spec.namespace.name for gateway %s/%s", gw.Namespace, gw.Name)
```

Comment:
```suggestion
		return nil, fmt.Errorf("invalid contour.spec.namespace.name for gateway %s/%s, it must be the same namespace as the Gateway", gw.Namespace, gw.Name)
```
I think it should be clarified why it was invalid.